auto_round/utils.py

```diff
@@ -1554,3 +1557,87 @@ def check_seqlen_compatible(input_seqlen, tokenizer=None, model=None):
         raise ValueError(f"seqlen({input_seqlen}) exceeds tokenizer.model_max_length({tokenizer.model_max_length}). " \
                 "Please oncider Consider lowering the '--seqlen' or increasing tokenizer.model_max_length.")
 
+def _use_more_bits(i_layer: int, n_layer: int):
+    return (i_layer < n_layer // 8) or (i_layer >= 7 * n_layer // 8) or ((i_layer - n_layer//8) % 3 == 2)
+
+def _get_digital_in_layer_name(layer_name):
+    patten = re.compile("([a-zA-Z]+\.){1,}(\d+)")
+    res = re.search(patten, layer_name)
+    if res:
+        return int(res[2])
+    else:
+        return None
+
+def get_layer_config_by_gguf_format(layer_config, gguf_format, model):
+    # TODO: support for other format later
+    if "q4_k_m" not in gguf_format:
+        return layer_config
+
+    import gguf
+    from auto_round.export.export_to_gguf.convert import Model
```

Comment:
It's better to return a dictionary to store the configuration (e.g., (q4_k_s, q5_k_s)) for each layer, which will help facilitate automatic mixed-bit quantization later.